### ISSUE TITLE
`chezmoi secret keyring get` dumps a stack trace

### ISSUE THREAD

**Issue by VorpalBlade** [twpayne/chezmoi#3451]
## Describe the bug

```console
$ chezmoi secret keyring get --service=a --user=b
panic: "get": no annotations

goroutine 1 [running]:
github.com/twpayne/chezmoi/v2/internal/cmd.getAnnotations(0xc000386f00)
	/build/chezmoi/src/chezmoi-2.42.3/internal/cmd/annotation.go:47 +0x13a
github.com/twpayne/chezmoi/v2/internal/cmd.(*Config).persistentPreRunRootE(0xc00041e000, 0xc000386f00, {0x55bb23684c9b?, 0xc0003b0300?, 0xc000386f00?})
	/build/chezmoi/src/chezmoi-2.42.3/internal/cmd/config.go:1886 +0x35
github.com/spf13/cobra.(*Command).execute(0xc000386f00, {0xc00033ede0, 0x2, 0x2})
	/build/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:954 +0x94b
github.com/spf13/cobra.(*Command).ExecuteC(0xc000005500)
	/build/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
	/build/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039
github.com/twpayne/chezmoi/v2/internal/cmd.(*Config).execute(0xc00071fd28?, {0xc00013e130, 0x5, 0x5})
	/build/chezmoi/src/chezmoi-2.42.3/internal/cmd/config.go:1216 +0x78
github.com/twpayne/chezmoi/v2/internal/cmd.runMain({{0x55bb249d0870, 0x6}, {0x0, 0x0}, {0x55bb249d19f0, 0x14}, {0x0, 0x0}}, {0xc00013e130, 0x5, ...})
	/build/chezmoi/src/chezmoi-2.42.3/internal/cmd/cmd.go:298 +0x25f
github.com/twpayne/chezmoi/v2/internal/cmd.Main({{0x55bb249d0870, 0x6}, {0x0, 0x0}, {0x55bb249d19f0, 0x14}, {0x0, 0x0}}, {0xc00013e130, 0x5, ...})
	/build/chezmoi/src/chezmoi-2.42.3/internal/cmd/cmd.go:111 +0xa5
main.main()
	/build/chezmoi/src/chezmoi-2.42.3/main.go:26 +0xfb
$ chezmoi secret keyring get
panic: "get": no annotations

goroutine 1 [running]:
github.com/twpayne/chezmoi/v2/internal/cmd.getAnnotations(0xc00033af00)
	/build/chezmoi/src/chezmoi-2.42.3/internal/cmd/annotation.go:47 +0x13a
github.com/twpayne/chezmoi/v2/internal/cmd.(*Config).persistentPreRunRootE(0xc00041c000, 0xc00033af00, {0x55b2bb2c7c9b?, 0xc000348300?, 0xc00033af00?})
	/build/chezmoi/src/chezmoi-2.42.3/internal/cmd/config.go:1886 +0x35
github.com/spf13/cobra.(*Command).execute(0xc00033af00, {0x55b2bd411360, 0x0, 0x0})
	/build/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:954 +0x94b
github.com/spf13/cobra.(*Command).ExecuteC(0xc00039d800)
	/build/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
	/build/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039
github.com/twpayne/chezmoi/v2/internal/cmd.(*Config).execute(0xc00071fd28?, {0xc00012c050, 0x3, 0x3})
	/build/chezmoi/src/chezmoi-2.42.3/internal/cmd/config.go:1216 +0x78
github.com/twpayne/chezmoi/v2/internal/cmd.runMain({{0x55b2bc613870, 0x6}, {0x0, 0x0}, {0x55b2bc6149f0, 0x14}, {0x0, 0x0}}, {0xc00012c050, 0x3, ...})
	/build/chezmoi/src/chezmoi-2.42.3/internal/cmd/cmd.go:298 +0x25f
github.com/twpayne/chezmoi/v2/internal/cmd.Main({{0x55b2bc613870, 0x6}, {0x0, 0x0}, {0x55b2bc6149f0, 0x14}, {0x0, 0x0}}, {0xc00012c050, 0x3, ...})
	/build/chezmoi/src/chezmoi-2.42.3/internal/cmd/cmd.go:111 +0xa5
main.main()
	/build/chezmoi/src/chezmoi-2.42.3/main.go:26 +0xfb
```

Doesn't seem to matter what parameters I provide to that specific command, it just dumps a stack trace.

## To reproduce

`chezmoi secret keyring get`

## Expected behavior

Either it should do something (retrieve a password from the DBUS secret service, which is GNOME keyring), or it should give a "nice" error message about what I'm doing wrong (instead of just dumping a stack).

## Output of command with the `--verbose` flag

```console
$ chezmoi --verbose secret keyring get
panic: "get": no annotations

goroutine 1 [running]:
github.com/twpayne/chezmoi/v2/internal/cmd.getAnnotations(0xc0004ccf00)
	/build/chezmoi/src/chezmoi-2.42.3/internal/cmd/annotation.go:47 +0x13a
github.com/twpayne/chezmoi/v2/internal/cmd.(*Config).persistentPreRunRootE(0xc00038a000, 0xc0004ccf00, {0x55fd973b3c9b?, 0xc00054e500?, 0xc0004ccf00?})
	/build/chezmoi/src/chezmoi-2.42.3/internal/cmd/config.go:1886 +0x35
github.com/spf13/cobra.(*Command).execute(0xc0004ccf00, {0xc00039a940, 0x1, 0x1})
	/build/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:954 +0x94b
github.com/spf13/cobra.(*Command).ExecuteC(0xc000005b00)
	/build/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
	/build/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039
github.com/twpayne/chezmoi/v2/internal/cmd.(*Config).execute(0xc00066dd28?, {0xc0000400b0, 0x4, 0x4})
	/build/chezmoi/src/chezmoi-2.42.3/internal/cmd/config.go:1216 +0x78
github.com/twpayne/chezmoi/v2/internal/cmd.runMain({{0x55fd986ff870, 0x6}, {0x0, 0x0}, {0x55fd987009f0, 0x14}, {0x0, 0x0}}, {0xc0000400b0, 0x4, ...})
	/build/chezmoi/src/chezmoi-2.42.3/internal/cmd/cmd.go:298 +0x25f
github.com/twpayne/chezmoi/v2/internal/cmd.Main({{0x55fd986ff870, 0x6}, {0x0, 0x0}, {0x55fd987009f0, 0x14}, {0x0, 0x0}}, {0xc0000400b0, 0x4, ...})
	/build/chezmoi/src/chezmoi-2.42.3/internal/cmd/cmd.go:111 +0xa5
main.main()
	/build/chezmoi/src/chezmoi-2.42.3/main.go:26 +0xfb
```

## Output of `chezmoi doctor`

<details>

```console
$ chezmoi doctor
RESULT    CHECK                       MESSAGE
warning   version                     v2.42.3, built at 2023-12-17T22:13:13Z
ok        latest-version              v2.42.3
ok        os-arch                     linux/amd64 (Arch Linux)
ok        uname                       Linux athena 6.6.7-zen1-1-zen #1 ZEN SMP PREEMPT_DYNAMIC Thu, 14 Dec 2023 03:45:20 +0000 x86_64 GNU/Linux
ok        go-version                  go1.21.5 (gc)
ok        executable                  /usr/bin/chezmoi
ok        config-file                 ~/.config/chezmoi/chezmoi.yaml, last modified 2023-12-14T22:41:08+01:00
ok        source-dir                  ~/.local/share/chezmoi is a git working tree (clean)
ok        suspicious-entries          no suspicious entries
ok        working-tree                ~/.local/share/chezmoi is a git working tree (clean)
ok        dest-dir                    ~ is a directory
ok        umask                       022
ok        cd-command                  found /bin/zsh
ok        cd-args                     /bin/zsh
info      diff-command                not set
ok        edit-command                found /usr/bin/code
ok        edit-args                   code -w
ok        git-command                 found /usr/bin/git, version 2.43.0
ok        merge-command               found /usr/bin/bcompare
ok        shell-command               found /bin/zsh
ok        shell-args                  /bin/zsh
info      age-command                 age not found in $PATH
ok        gpg-command                 found /usr/bin/gpg, version 2.4.3
info      pinentry-command            not set
info      1password-command           op not found in $PATH
info      bitwarden-command           bw not found in $PATH
info      bitwarden-secrets-command   bws not found in $PATH
info      dashlane-command            dcli not found in $PATH
info      doppler-command             doppler not found in $PATH
info      gopass-command              gopass not found in $PATH
ok        keepassxc-command           found /usr/bin/keepassxc-cli, version 2.7.6
info      keepassxc-db                not set
info      keeper-command              keeper not found in $PATH
info      lastpass-command            lpass not found in $PATH
info      pass-command                pass not found in $PATH
info      passhole-command            ph not found in $PATH
info      rbw-command                 rbw not found in $PATH
info      vault-command               vault not found in $PATH
info      vlt-command                 vlt not found in $PATH
info      secret-command              not set
```

</details>

## Additional context

Binary is from Arch Linux archives. But I also tried the upstream binary on my raspberry pi and I get the exact same issue.

**Comment by VorpalBlade** [comment]
Note: "chezmoi secret keyring set" does the exact same thing, as does the "delete" subcommand. None of them seem to work in fact.

**Comment by twpayne** [comment]
This was fixed by @halostatue in #3425, but not released yet.

It's time for a new release anyhow, so I'll do one now.
